viite-backend/api/src/main/scala/fi/liikennevirasto/digiroad2/ApiUtils.scala

```diff
@@ -50,20 +57,21 @@ object ApiUtils {
     val objectExists = s3Service.isS3ObjectAvailable(s3Bucket, workId, 2, Some(objectTTLSeconds))
 
     (params.get("retry"), objectExists) match {
-      case (_, true) =>
+      case (_, true) => // the result is ready, make the last redirect to the resulting file
```

Comment:
jos energiaa nää komnettit vois kyl muuntaa Tila olioksi hyvin nimetyillä arvoilla joita sitten mach käydään läpi,

Reply:
Een tiedä...
Jättäisin kommenteista silti toimintakuvaukset, jolloin saisin jotain suurinpiirtein tällaista:

val resultReady = (_, true)
val firstCall = (None, false)
val otherwiseRedirect = (Some(retry: String), false)

case resultReady => //  make the last redirect to the resulting file
  ---
case firstCall => // get the API call running
  ---
case otherwiseRedirect => // go to the redirect loop (or decide we have tried too many times)
  ---
  
En oikein näe tämän hyötyjä. Mutta vakuuttakaa minut!

Reply:
(Johan huutaa isolla fontilla... 😬 )

Reply:
itse en ihan ajatellut tossa mielessä

Reply:
class.Apply(boolean pametrit)  {
case resultReady => tee jotakin
case firstCall => tee jotakin 
case otherwiseRedirect => tee jotakin
} 
sitten ajatussena olis  mietitään noille hyvät nimet joka siten korvaisi noi kommentit, kommentti vanhenee kun

Reply:
muutenkin toi nimi voi olla pitkä jos on tarpeellista,

Reply:
mut joo tää on kyl aika miten haluu tehdä asioita homma